bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableTable.java

```diff
@@ -298,8 +302,24 @@ public ResultScanner getScanner(final Scan scan) throws IOException {
     LOG.trace("getScanner(Scan)");
     Span span = TRACER.spanBuilder("BigtableTable.scan").startSpan();
     try (Scope scope = TRACER.withSpan(span)) {
+      ResultScanner scanner;
+      if (scan.getCaching() == -1) {
+        scanner = clientWrapper.readRows(hbaseAdapter.adapt(scan));
+      } else {
+        final Query request = hbaseAdapter.adapt(scan);
+        final RequestContext requestContext =
+            RequestContext.create("ProjectId", "InstanceId", "AppProfile");
+        int requestedPageSize = (int) request.toProto(requestContext).getRowsLimit();
+        if (requestedPageSize > scan.getCaching()) {
```

Comment:
I'm not sure why line 313 to line 317 is necessary. This should be handled by QueryPaginator already. And why do we still have a default PAGE_SIZE ? PAGE_SIZE should always be value from scan.getCaching().

Reply:
Done